File: frontend/src/types/Order.ts
import type { Address } from './Address';
import type { Product } from './catalog.types';
import type { User } from './auth.types';

export interface OrderItem {
    id: number;
    product: Product;
    quantity: number;
    priceAtPurchase: number;
}

export interface PaymentDetails {
    id: number;
    paymentMethod: string;
    transactionId: string;
    status: string;
    paymentDate: string;
}

export interface Order {
    id: number;
    user: User;
    shippingAddress: Address;
    paymentDetails: PaymentDetails;
    orderItems: OrderItem[];
    totalAmount: number;
    totalItems: number;
    status: string;
    couponCode?: string;
    discountAmount?: number;
    trackingNumber?: string;
    carrier?: string;
    createdAt: string;
}

File: backend/src/main/java/com/mrsdeores/models/Order.java
package com.mrsdeores.models;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "address_id", nullable = false)
    private Address shippingAddress;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "payment_details_id", referencedColumnName = "id")
    private PaymentDetails paymentDetails;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<OrderItem> orderItems = new ArrayList<>();

    @Column(name = "total_amount", nullable = false)
    private BigDecimal totalAmount;

    @Column(name = "total_items")
    private Integer totalItems;

    @Column(nullable = false)
    private String status; // "CREATED", "MOCK_PAID", "DELIVERED"

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public void addOrderItem(OrderItem item) {
        orderItems.add(item);
        item.setOrder(this);
    }
}
